### DISCUSSION TITLE
The bot does not work with the imported router

### DISCUSSION THREAD

**Discussion by vanusha1991** [aiogram/aiogram#1346] [Q&A]
### Checklist

- [x] I am sure the error is coming from aiogram code
- [X] I have searched in the issue tracker for similar bug reports, including closed ones

### Operating system

Windows 10

### Python version

3.10.4

### aiogram version

3.1.1

### Expected behavior

The bot will start working with the router.

### Current behavior

The bot does not respond to events.

### Steps to reproduce

Did the following:
 
01) Launched the bot from the dispatcher (with handler registration) - the bot is working ✅
02) Launched the bot from the root router (without registering a handler) - the bot works ✅
03) Launched the bot from the root router (with handler registration) - the bot works ✅
04) Launched a bot with middleware - the bot is working ✅
05) Launched a bot with middleware + root router (without registering a handler) - the bot works ✅
06) Launched a bot with middleware + root router (with handler registration) - the bot works, but does not respond to the ChatMemberUpdated event ❗️
07) Launched the bot from the root router (without registering a handler) + middleware - the bot works ✅
08) I launched the bot with a root router (with handler registration) + middleware - the bot works, but does not respond to the ChatMemberUpdated event ❗️
09) Launched a bot with middleware + imported router (without handler registration) - the bot works ✅
10) Launched a bot with middleware + imported router (with handler registration) - the bot does not respond to events ❌
11) Launched the bot with an imported router (without registering a handler) + middleware - the bot works ✅
12) Launched a bot with an imported router (with handler registration) + middleware - the bot does not respond to events ❌
13) Launched a bot with middleware + root router (without handler registration) + imported router (without handler registration) - the bot works ✅
14) Launched a bot with middleware + root router (with handler registration) + imported router (with handler registration) - the bot works, but does not respond to the ChatMemberUpdated event ❗️
15) Launched a bot with middleware + imported router (without handler registration) + root router (without handler registration) - the bot works ✅
16) Launched a bot with middleware + imported router (with handler registration) + root router (with handler registration) - the bot works, but does not respond to the ChatMemberUpdated event ❗️

### Code example

```python3
main module https://pastebin.com/hekMcdca
router module https://pastebin.com/3eTgfLJ3
middleware module https://pastebin.com/6KJEKw2D
```

### Logs

```sh
No error in logs
```

**Comment by JrooTJunior** [comment]
BotStatus middleware is incorrect. You should call await handler(event, data) to handle this event.

**Comment by vanusha1991** [comment]
> BotStatus middleware is incorrect. You should call await handler(event, data) to handle this event.


BotStatus this middleware simply displays a notification in the log that the bot is blocked or unblocked.
With this event, no further processing is required.
Therefore, it does not have await handler(event, data)

**Comment by JrooTJunior** [comment]
1. Provided code is not an minimum reproducible example and can't be copy-pasted and executed locally. If you really want help - you should provide code example that can be used not only by you.
1. Middlewares are totally incorrect because should always returns result of the next handler (`return await handler(event,data)`). If you don't call handler and/or return the result - event will not be propagated to the any handler or to nested routers.
3. Middlewares will be called only if you have an any event handler for the same event type as middleware mounded, but you don't have any `my_chat_member` event handler registered, if you want to explicitly enable unused event types - you should configure manually allowed_updates property of the webhook or polling process.
4. If you want to only log that the bot is blocked or unblocked - you should use my_chat_member handler instead of middleware.
